stdlib/http/src/main/java/org/ballerinalang/net/http/ClientHandshakeListener.java

```diff
@@ -0,0 +1,116 @@
+                                   CountDownLatch countDownLatch) {
+        super(webSocketClient, wsService, clientConnectorListener, readyOnConnect, countDownLatch);
+        this.webSocketClient = webSocketClient;
+        this.wsService = wsService;
+        this.clientConnectorListener = clientConnectorListener;
+        this.readyOnConnect = readyOnConnect;
+        this.countDownLatch = countDownLatch;
+    }
+
+    @Override
+    public void onSuccess(WebSocketConnection webSocketConnection, HttpCarbonResponse carbonResponse) {
+        //Using only one service endpoint in the client as there can be only one connection.
+        webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD, HttpUtil.createResponseStruct(carbonResponse));
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, webSocketConnection, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
+        clientConnectorListener.setConnectionInfo(connectionInfo);
+        webSocketClient.set(WebSocketConstants.CLIENT_CONNECTOR_FIELD, webSocketConnector);
+        if (hasRetryConfig(webSocketClient)) {
+            if (readyOnConnect) {
+                webSocketConnection.readNextFrame();
+            }
+            WebSocketUtil.populateEndpoint(webSocketConnection, webSocketClient);
+            logger.info(CONNECTED_TO + webSocketClient.getStringValue(WebSocketConstants.
+                    CLIENT_URL_CONFIG));
+        } else {
+            RetryContext retryConfig = (RetryContext) webSocketClient.getNativeData(RETRY_CONFIG);
+            logger.info(CONNECTED_TO + webSocketClient.getStringValue(WebSocketConstants.
+                    CLIENT_URL_CONFIG));
+            setWebSocketEndpoint(retryConfig, webSocketClient, webSocketConnection);
+            if (retryConfig.isConnectionMade() || readyOnConnect) {
+                webSocketConnection.readNextFrame();
+            }
+            setReconnectContexValue(retryConfig);
+        }
+        countDownLatch.countDown();
+    }
+
+    @Override
+    public void onError(Throwable throwable, HttpCarbonResponse response) {
+        if (response != null) {
+            webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD, HttpUtil.createResponseStruct(response));
+        }
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, null, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
```

Comment:
There's a new change made at this point in a newer PR. Please note that as well: https://github.com/ballerina-platform/ballerina-lang/pull/18216